app/components/Registration/AccountRegistrationConfirm.jsx

```diff
@@ -0,0 +1,217 @@
+import AccountStore from "stores/AccountStore";
+import WalletDb from "stores/WalletDb";
+import notify from "actions/NotificationActions";
+import TransactionConfirmStore from "stores/TransactionConfirmStore";
+import Translate from "react-translate-component";
+import {FetchChain} from "bitsharesjs/es";
+import WalletUnlockActions from "actions/WalletUnlockActions";
+import Icon from "components/Icon/Icon";
+import ls from "common/localStorage";
+
+const STORAGE_KEY = "__graphene__";
+const ss = new ls(STORAGE_KEY);
+
+class AccountRegistrationConfirm extends React.Component {
+    static propTypes = {
+        accountName: PropTypes.string.isRequired,
+        password: PropTypes.string.isRequired,
+        toggleConfirmed: PropTypes.func.isRequired
+    };
+
+    static contextTypes = {
+        router: PropTypes.object.isRequired
+    };
+
+    constructor() {
+        super();
+        this.state = {
+            confirmed: false
+        };
+        this.onFinishConfirm = this.onFinishConfirm.bind(this);
+        this.toggleConfirmed = this.toggleConfirmed.bind(this);
+        this.createAccount = this.createAccount.bind(this);
+        this.onCreateAccount = this.onCreateAccount.bind(this);
+    }
+
+    shouldComponentUpdate(nextProps, nextState) {
+        return nextState.confirmed !== this.state.confirmed;
+    }
+
+    onFinishConfirm(confirmStoreState) {
+        if (
+            confirmStoreState.included &&
+            confirmStoreState.broadcasted_transaction
+        ) {
+            TransactionConfirmStore.unlisten(this.onFinishConfirm);
+            TransactionConfirmStore.reset();
+
+            FetchChain("getAccount", this.state.accountName).then(() => {
+                this.props.history.push(
+                    "/wallet/backup/create?newAccount=true"
+                );
+            });
+        }
+    }
+
+    onCreateAccount(e) {
+        e.preventDefault();
+        this.createAccount(this.props.accountName, this.props.password);
+    }
+
+    createAccount(name, password) {
+        const refcode = this.refs.refcode ? this.refs.refcode.value() : null;
+        const {referralAccount} = AccountStore.getState();
+
+        AccountActions.createAccountWithPassword(
+            name,
+            password,
+            this.state.registrarAccount,
+            referralAccount || this.state.registrarAccount,
+            0,
+            refcode
+        )
+            .then(() => {
+                AccountActions.setPasswordAccount(name);
+                if (this.state.registrarAccount) {
+                    FetchChain("getAccount", name).then(() => {
+                        this.unlockAccount(name, password);
+                    });
+                    TransactionConfirmStore.listen(this.onFinishConfirm);
+                } else {
+                    FetchChain("getAccount", name).then(() => {});
+                    this.unlockAccount(name, password);
+                    ss.set("isAuthentificated", true);
```

Comment:
The same comment about Storages/Actions.

Reply:
Fixed.